File: 0643_max_average_subarray_i/max_average_sub_array_fixed_length.py
from typing import List


class MaxAverageSubArrayFixedLength:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        n = len(nums)
        max_sum = window_sum = sum(nums[:k])
        start = 0

        for i in range(k, n):
            window_sum += nums[i] - nums[i-k]

            if window_sum > max_sum:
                  max_sum = window_sum
                  start = i - k + 1
        
        print(nums[start:start+k])
                  
        return max_sum/k   


solution = MaxAverageSubArrayFixedLength()

nums = [1, 12, -5, -6, 50, 3]
k = 4

print(nums)
answer = solution.findMaxAverage(nums, k)

print(f"Answer : {answer}");

File: 0643_max_average_subarray_i/max_average_sub_array_fixed_length.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var findMaxAverage = function (nums, k) {
    let first_array = nums.slice(0, k);
    let window_sum = first_array.reduce((a, c) => a + c, 0);

    let max_sum = window_sum;
    let start = 0;

    for (let i = k; i < nums.length; i++) {
        window_sum += nums[i] - nums[i - k];

        if (window_sum > max_sum) {
            max_sum = window_sum;
            start = i - k + 1;
        }
    }

    for (let i = start; i < start + k; i++) {
        console.log(`nums[${i}] = ${nums[i]}`)
    }

    return max_sum / k;
};

let nums = [1, 12, -5, -6, 50, 3];
let k = 4;

console.log(nums);
let answer = findMaxAverage(nums, k);

console.log(`Answer = ${answer}`);